### PR TITLE
feat(tooltip, toggletip, menu): add anchor slot (VIV-1539)

### PR DESCRIPTION
**Pull request by RichardHelm**
Sets `display: contents` only when using the slot to maintain backwards compatibility

### CODE REVIEW

File: libs/components/src/lib/menu/README.md [hunk 1]
```diff
@@ -8,10 +8,20 @@ Menu, commonly known as a "context menu", is an element that is displayed upon u
 </script>
 ```
 
+## Anchoring
```
Comment (rachelbt) [inline libs/components/src/lib/menu/README.md:11]: I'm not sure this needs to be on the top.
Isn't it enough as a slot?

File: libs/components/src/lib/popup/partials/variables.scss [hunk 2]
```diff
@@ -1 +1,3 @@
 $popup-width: --_popup-width;
+$popup-display: --popup-display;
+$popup-z-index: --popup-z-index;
```
Comment (rachelbt) [inline libs/components/src/lib/popup/partials/variables.scss:3]: these too are mainly for our internal use - right?
if so - I think they need to be with low-dash as well:
$popup-display: --_popup-display;
$popup-z-index: --_popup-z-index;

**Review by rachelbt — COMMENTED** [review]
NICE!
I don't see a reason for the anchored to be in the top of the doc page.
More over - I think that all that info needs to be under slot- anchor, and maybe we should add about the anchor as a property that this is not the best way to anchor and send people to use the slot (this way we can one day deprecate that).
Not sure about the image that changes - seems not relevant

File: libs/components/src/lib/menu/README.md [hunk 3]
```diff
@@ -115,6 +80,40 @@ Use the `placement` attribute to control the position of the menu relative to it
 </script>
 ```
 
+### Anchor
+
+As an alternative to using the `anchor` slot, you can use the `anchor` member to set the anchor.
```
Comment (rachelbt) [inline libs/components/src/lib/menu/README.md:85]: its better to use the slot - right?
So why won't we say something like:
It is best to use the anchor slot (link) but you can also....

**Review by rachelbt — COMMENTED** [review]
left a comment for menu.- but its for all the others as well :)

File: libs/components/src/shared/patterns/anchored.scss [hunk 4]
```diff
@@ -0,0 +1,9 @@
+@use "../../lib/popup/partials/variables" as variables;
+
+:host {
+	&[slotted-anchor] {
+		#{variables.$popup-display}: contents;
+	}
```
Comment (rachelbt) [inline libs/components/src/shared/patterns/anchored.scss:6]: this is not working...
this needs to be : &([slotted-anchor]) {...}
but this throws error.
maybe this is better:
```
:host {
	display: var(#{variables.$popup-display}, inline);
}

:host([slotted-anchor]) {
	#{variables.$popup-display}: contents;
}

```

Comment (RichardHelm) [reply libs/components/src/shared/patterns/anchored.scss]: Good catch 👍

**Review by rachelbt — APPROVED** [review]
NICE!

**Pull request merged**